compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1116,24 +1116,87 @@ private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, STNode visibilityQual
         return funcDefOrType;
     }
 
-    private STNode parseFunctionKeywordRhs(STNode metadata, STNode visibilityQualifier, List<STNode> qualifiers,
-                                           STNode functionKeyword, boolean isObjectMember, boolean isObjectTypeDesc) {
+    private STNode parseFunctionDefinition(STNode metadata, STNode visibilityQualifier, STNode resourcePath,
+                                           List<STNode> qualifiers, STNode functionKeyword, boolean isObjectMember,
+                                           boolean isObjectTypeDesc, STNode name) {
+        switchContext(ParserRuleContext.FUNC_DEF);
+        STNode funcSignature = parseFuncSignature(false);
+        STNode funcDef = parseFuncDefOrMethodDeclEnd(metadata, visibilityQualifier, qualifiers, functionKeyword, name,
+                                                     resourcePath, funcSignature, isObjectMember, isObjectTypeDesc);
+        endContext();
+        return funcDef;
+    }
+
+    /**
+     * Parse function definition or function type descriptor identifier rhs.
+     *
+     * @param metadata Preceding metadata
+     * @param visibilityQualifier Preceding visibility qualifier
+     * @param qualifiers Preceding visibility qualifier
+     * @param functionKeyword `function` keyword
+     * @param isObjectMember Whether object member or not
+     * @param isObjectTypeDesc Whether object type or not
+     * @return Parsed node
+     */
+    private STNode parseFuncDefOrFuncTypeDescRhs(STNode name, STNode metadata, STNode visibilityQualifier,
+                                              List<STNode> qualifiers, STNode functionKeyword, boolean isObjectMember,
```

Comment:
Can we order the parameters to be aligned with the syntax order?. i.e: the `name` parameter should come after `functionKeyword`. Please change in other methods as well